web/courses_test.go

```diff
@@ -94,7 +92,7 @@ func TestNewCourseExistingRepos(t *testing.T) {
 	if course != nil {
 		t.Fatal("expected CreateCourse to fail with AlreadyExists")
 	}
-	if err != nil && status.Code(err) != codes.FailedPrecondition {
+	if err != nil && connect.CodeOf(err) != connect.CodeFailedPrecondition {
```

Comment:
The `qfService.CreateCourse()` will actually return the `failedPrecondition` code in three different situations:

```
if contextCanceled(ctx) {
			return nil, connect.NewError(connect.CodeFailedPrecondition, ErrContextCanceled)
		}
```

and

```
if err == ErrAlreadyExists || err == ErrFreePlan {
			return nil, connect.NewError(connect.CodeFailedPrecondition, err)
		}
```
Not sure if `CodeAlreadyExists` is a reasonable code to return in case of a free plan. Should we just separate these two cases?

Reply:
Agree that we should separate `ErrAlreadyExists` and `ErrFreePlan`. As for the `contextCanceled`, I don't think it makes sense to use `FailedPrecondition`. See #808.

Reply:
Changed `ErrAlreadyExists` to return the `CodeAlreadyExists`. I was thinking to limit changes to the context errors to a separate PR though.

Reply:
ok. Sounds good to me!